pkg/envoy/admin/kds_client_test.go

```diff
@@ -114,16 +115,17 @@ var _ = Describe("KDS client", func() {
 
 		// and
 		request := <-stream.receivedRequests
-		err := streams.ResponseReceived(zoneName, &mesh_proto.XDSConfigResponse{
-			RequestId: request.RequestId,
-			Result: &mesh_proto.XDSConfigResponse_Error{
-				Error: "failed",
-			},
-		})
+		Eventually(func() error {
+			return streams.ResponseReceived(zoneName, &mesh_proto.XDSConfigResponse{
+				RequestId: request.RequestId,
+				Result: &mesh_proto.XDSConfigResponse_Error{
+					Error: "failed",
+				},
+			})
+		}, "10s", "100ms").Should(Succeed())
 
 		// then
-		Expect(err).ToNot(HaveOccurred())
-		err = <-errCh
+		err := <-errCh
 		Expect(err).To(MatchError("error response from Zone CP: failed"))
```

Comment:
As in an edge case it can block the whole suite forever, maybe instead you could do:
```suggestion
		Eventually(errCh).Should(Receive(MatchError("error response from Zone CP: failed")))
```